### PR TITLE
[SYSTEMML-568] Frame MLContext support

### PR DESCRIPTION
**Pull request by asurve**
This change is based on GNMFTest code need. FrameTest.java has added for similar support as GNMFTest. So this matches older way of MLContext api calls. To fit into new approach additional changes will be needed. I will work on new changes soon.

### CODE REVIEW

**Comment by niketanpansare** [comment]
LGTM. Thanks @asurve for adding frame support for MLContext :+1:

**Comment by asurve** [comment]
Thanks Niketan for reviewing changes. 
Glen, I retrofitted these changes with my changes. 
Thanks
Arvind

File: src/main/java/org/apache/sysml/api/mlcontext/MLContextConversionUtil.java [hunk 1]
```diff
@@ -187,6 +189,38 @@ public static MatrixObject matrixBlockToMatrixObject(String variableName, Matrix
...
+	 * 
+	 * @param variableName
+	 *            name of the variable associated with the frame
+	 * @param frameBlock
+	 *            frame as a FrameBlock
+	 * @param matrixMetadata
+	 *            the matrix metadata
+	 * @return the {@code FrameBlock} converted to a {@code FrameObject}
+	 */
+	public static FrameObject frameBlockToframeObject(String variableName, FrameBlock frameBlock,
```
Comment (deroneriksson) [inline src/main/java/org/apache/sysml/api/mlcontext/MLContextConversionUtil.java]: Hi @asurve. Please update line 202 "frameBlockToframeObject" to "frameBlockToFrameObject". Thanks!

File: src/main/java/org/apache/sysml/api/mlcontext/MLContextConversionUtil.java [hunk 2]
```diff
@@ -187,6 +189,38 @@ public static MatrixObject matrixBlockToMatrixObject(String variableName, Matrix
...
+			}
+			MatrixFormatMetaData mtd = new MatrixFormatMetaData(matrixCharacteristics,
+					OutputInfo.BinaryBlockOutputInfo, InputInfo.BinaryBlockInputInfo);
+			FrameObject frameObject = new FrameObject(MLContextUtil.scratchSpace() + "/"
+					+ variableName, mtd);
+			frameObject.acquireModify(frameBlock);
+			frameObject.release();
+			return frameObject;
+		} catch (CacheException e) {
+			throw new MLContextException("Exception converting MatrixBlock to MatrixObject", e);
```
Comment (deroneriksson) [inline src/main/java/org/apache/sysml/api/mlcontext/MLContextConversionUtil.java]: Please update line 219 MatrixBlock to FrameBlock and MatrixObject to FrameObject.

File: src/main/java/org/apache/sysml/api/mlcontext/MLContextUtil.java [hunk 3]
```diff
@@ -456,6 +457,11 @@ public static Data convertInputType(String parameterName, Object parameterValue,
 			MatrixObject matrixObject = MLContextConversionUtil.matrixBlockToMatrixObject(name, matrixBlock,
 					matrixMetadata);
 			return matrixObject;
+		} else if (value instanceof FrameBlock) {
+			FrameBlock frameBlock = (FrameBlock) value;
+			FrameObject frameObject = MLContextConversionUtil.frameBlockToframeObject(name, frameBlock,
```
Comment (deroneriksson) [inline src/main/java/org/apache/sysml/api/mlcontext/MLContextUtil.java]: frameBlockToframeObject to frameBlockToFrameObject

**Pull request closed without merging**